src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,111 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import {
+  getPostDetails,
+  getPostComments,
+  addComment,
+  deleteComment,
+} from '../../api/api';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type CommentInfo = Omit<Comment, 'id'>;
+
+type Props = {
+  selectedPostId: number,
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [selectedPost, selectPost] = useState<Post>();
+  const [postComments, setComments] = useState<Comment[]>();
+  const [isCommentsHidden, hideComments] = useState<boolean>(false);
+
+  const getPost = async () => {
+    try {
+      const [post, comments] = await Promise.all([
+        getPostDetails(selectedPostId),
+        getPostComments(selectedPostId),
+      ]);
+
+      if (post) {
+        selectPost(post);
+      }
+
+      if (comments) {
+        setComments(comments);
+      }
+    } catch {
+      throw new Error('Data was\'t found');
+    }
+  };
+
+  const addPostComment = (comment: CommentInfo) => {
+    addComment(comment);
+    getPost();
```

Comment:
don't need load post details and all comments, you can add your new comment to postComments arr if the method addComment don't catch err